scripts/globals/chocobo.lua

```diff
@@ -118,8 +150,21 @@ xi.chocobo.renterOnEventFinish = function(player, csid, option, eventSucceed)
         local mLvl   = player:getMainLvl()
         local zoneId = player:getZoneID()
         local info   = chocoboInfo[zoneId]
+        local trade  = player:getLocalVar('Chocopass')
+        local duration = 900
 
-        if info then
+        if not info then
+            printf('[warning] player %s passed bad zoneId %i in xi.chocobo.renterOnEventFinish', player:getName(), zoneId)
+            return
+        end
+
+        if trade == 1 then-- If the player used a Chocopass/Chocobo Ticket
+            player:setLocalVar('Chocopass', 0)
+            player:setLocalVar('ChocopassDuration', 0)
+            player:tradeComplete()
+
+            duration = player:getLocalVar('ChocopassDuration')
```

Comment:
I think this needs to happen before line 163

Reply:
Bah, thanks for spotting that